sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_connection.py

```diff
@@ -507,10 +510,13 @@ def _incoming_end(self, channel, frame):
         """
         try:
             self._incoming_endpoints[channel]._incoming_end(frame)  # pylint:disable=protected-access
+            self._incoming_endpoints.pop(channel)
```

Comment:
Just curious: For an outgoing_close, could we receive a response from the service on that channel to confirm it is closing? (i.e from the incoming_endpoint channel). Or could we receive an error on outgoing_close that we would need the incoming channel to still be there for?

Reply:
I think if the entire `Connection` was being closed, then the state would be set to END after sending after sending a CLOSE to the other side. In this case while this channel has closed, there might be other channels still open. Unless I missed it here, doesnt seem to be the case in channels http://docs.oasis-open.org/amqp/core/v1.0/csprd01/amqp-core-transport-v1.0-csprd01.html#doc-idp191152

Reply:
It's actually a really good question - and why this was a TODO in the first place.
It's hard to know where to pop the channel/session/link because if it's initiated by us, we don't always wait around for the service response. 
So we need to test/be able to handle:
- If we get an incoming close/end/detach, then is this in response to one we already sent? If not, we can probably just send the response frame then pop it.
- If we are sending a close/end/detach, if we pop it then we need to be able to handle additional incoming frames that are sent to it. Otherwise, do we wait until we explicitly receive the response? That can be slow and messy if there's other things going on in the connection.